src/scos_tekrsa/hardware/tekrsa_sigan.py

```diff
@@ -182,83 +185,81 @@ def iq_bandwidth(self, iq_bandwidth):
+        self._iq_bandwidth, self._sample_rate = self.rsa.IQSTREAM_GetAcqParameters()
         msg = (
             "Set Tektronix RSA IQ Bandwidth: "
-            + f"{new_bw:.1f} Hz, resulting in sample rate: {new_sr:.1f} samples/sec"
+            + f"{self._iq_bandwidth:.1f} Hz, resulting in sample rate: "
+            + f"{self._sample_rate:.1f} samples/sec"
         )
         logger.debug(msg)
 
     @property
     def frequency(self):
-        return self.rsa.CONFIG_GetCenterFreq()
+        self._frequency = self.rsa.CONFIG_GetCenterFreq()
+        return self._frequency
 
     @frequency.setter
     def frequency(self, freq):
         """Set the device center frequency."""
         self.rsa.CONFIG_SetCenterFreq(freq)
-        msg = (
-            "Set Tektronix RSA center frequency: "
-            + f"{self.rsa.CONFIG_GetCenterFreq():.1f} Hz"
-        )
+        self._frequency = self.rsa.CONFIG_GetCenterFreq()
+        msg = f"Set Tektronix RSA center frequency: {self._frequency:.1f} Hz"
         logger.debug(msg)
 
     @property
     def reference_level(self):
-        return self.rsa.CONFIG_GetReferenceLevel()
+        self._reference_level = self.rsa.CONFIG_GetReferenceLevel()
+        return self._reference_level
 
     @reference_level.setter
     def reference_level(self, reference_level):
         """Set the device reference level."""
         self.rsa.CONFIG_SetReferenceLevel(reference_level)
-        msg = (
-            "Set Tektronix RSA reference level: "
-            + f"{self.rsa.CONFIG_GetReferenceLevel():.1f} dBm"
-        )
+        self._reference_level = self.rsa.CONFIG_GetReferenceLevel()
+        msg = f"Set Tektronix RSA reference level: {self._reference_level:.1f} dBm"
         logger.debug(msg)
 
     @property
     def attenuation(self):
         if self.device_name not in ["RSA306B", "RSA306"]:
-            return self.rsa.CONFIG_GetRFAttenuator()
+            # API returns attenuation as negative value. Convert to positive.
+            self._attenuation = -1 * self.rsa.CONFIG_GetRFAttenuator()
         else:
             logger.debug("Tektronix RSA 300 series device has no attenuator.")
-            return None
+            self._attenuation = None
+        return self._attenuation
 
     @attenuation.setter
     def attenuation(self, attenuation):
         """Set device attenuation, in dB, for RSA 500/600 series devices"""
         if self.device_name not in ["RSA306B", "RSA306"]:
             self.rsa.CONFIG_SetAutoAttenuationEnable(False)
+            # API requires attenuation set as a negative number. Convert to negative.
             self.rsa.CONFIG_SetRFAttenuator(
                 -1 * attenuation
             )  # rounded to nearest integer
-            msg = (
-                "Set Tektronix RSA attenuation: "
-                + f"{self.rsa.CONFIG_GetRFAttenuator():.1} dB"
-            )
-            logger.debug(msg)
+            self._attenuation = self.rsa.CONFIG_GetRFAttenuator()
```

Comment:
The property above does -1 * self.rsa.CONFIG_GetRFAttenuator()

Reply:
Good catch. Fixed in [d313a2b](https://github.com/NTIA/scos-tekrsa/pull/30/commits/d313a2b4a64a0b78464caeb817967995fe331496)